lib/pool.js

```diff
@@ -72,14 +77,14 @@ class Pool extends EventEmitter {
       }
     }
 
-    this[kOnConnect] = function onConnect (client) {
+    this[kOnConnect] = function onConnect (origin, clients) {
       pool[kConnected]++
-      pool.emit('connect', client)
+      pool.emit('connect', origin, [...clients, pool])
```

Comment:
Does this array always contain one `Client` and the `Pool`?

Reply:
Right now yes. But I envision that we could add a factory argument to the pool (similar to what we do for agent) and then it wouldn't be so sure.